core/includes/cache.inc

```diff
@@ -308,7 +308,7 @@ class BackdropDatabaseCache implements BackdropCacheInterface {
       // is used here only due to the performance overhead we would incur
       // otherwise. When serving an uncached page, the overhead of using
       // db_select() is a much smaller proportion of the request.
-      $result = db_query('SELECT cid, data, created, expire, serialized FROM {' . db_escape_table($this->bin) . '} WHERE cid IN (:cids)', array(':cids' => $cids));
+      $result = db_query('SELECT cid, data, created, expire, serialized FROM {' . db_escape_table($this->bin) . '} WHERE cid IN (:cids) AND (expire <= 0 OR expire > :time)', array(':cids' => $cids, ':time' => REQUEST_TIME));
```

Comment:
This amounts to a fairly significant API change. We've changed the way the page cache works that it will retrieve but ignore expired entries, so I don't think this is necessary at this point.